### PR TITLE
e2e: test brownfield setup

### PR DESCRIPTION
**Pull request by 3quanfeng**
## Description
- Add e2e test case to test brownfield scenario
- brownfield set up will install non-shared environment first and then reinstalled with shared enabled, non-blacklisted rule is expected to be wiped out, while blacklisted rules shall be retained

### CODE REVIEW

File: scripts/e2e/run-e2e.sh [hunk 1]
```diff
@@ -26,22 +26,66 @@ function InstallAGIC() {
     helm repo add staging https://appgwingress.blob.core.windows.net/ingress-azure-helm-package-staging/
     helm repo update
 
-    # AAD pod identity is taking time to assign identity. Timeout is set to 60 sec
+    # delete agic if exist
+    helm delete agic-${version} -n agic || true
```
Comment (akshaysngupta) [inline scripts/e2e/run-e2e.sh:30]: do we need to delete ?

File: scripts/e2e/run-e2e.sh [hunk 2]
```diff
@@ -26,22 +26,66 @@ function InstallAGIC() {
...
+    kubectl apply -f test-prohibit-backend.yaml
+
+}
+
+function SetupSharedBackend() {
+    
+    # delete agic with share disabled
+    helm delete agic-${version} -n agic || true
+
+    helm repo update
```
Comment (akshaysngupta) [inline scripts/e2e/run-e2e.sh:57]: no need to upgrade ?

File: scripts/e2e/test-prohibit-backend.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+kind: Ingress
+metadata:
+  name: brownfield-blacklist-ingress
+  namespace: test-brownfield-ns
+  annotations:
+    kubernetes.io/ingress.class: azure/application-gateway
+    appgw.ingress.kubernetes.io/backend-path-prefix: "/"
+spec:
+  rules:
+    - host: brownfield-blacklist-ns.host
```
Comment (akshaysngupta) [inline scripts/e2e/test-prohibit-backend.yaml:55]: host here doesn't match the hostname in the above .yaml

brownfield-blacklist.host != brownfield-blacklist-ns.host

File: scripts/e2e/run-e2e.sh [hunk 4]
```diff
@@ -26,22 +26,66 @@ function InstallAGIC() {
...
+    --set armAuth.identityResourceID=${identityResourceId} \
+    --set armAuth.identityClientID=${identityClientId} \
+    --set rbac.enabled=true \
+    --timeout 120s \
+    --wait \
+    -n agic \
+    --version ${version}
+
+    # create namespace for testing shared backend
+    kubectl create namespace test-brownfield-ns || true
```
Comment (akshaysngupta) [inline scripts/e2e/run-e2e.sh:45]: create the namespace as part of the yaml ?

File: scripts/e2e/run-e2e.sh [hunk 5]
```diff
@@ -26,22 +26,66 @@ function InstallAGIC() {
...
+
+    # apply backends to test both non-prohibited and prohibited target
+    kubectl apply -f test-prohibit-backend.yaml
+
+}
+
+function SetupSharedBackend() {
+    
+    # delete agic with share disabled
+    helm delete agic-${version} -n agic || true
```
Comment (akshaysngupta) [inline scripts/e2e/run-e2e.sh:55]: do we need to delete ? Can you just update the current release ?

File: scripts/e2e/run-e2e.sh (hunk 2 above)
Comment (3quanfeng) [reply scripts/e2e/run-e2e.sh]: removed

File: scripts/e2e/run-e2e.sh (hunk 4 above)
Comment (3quanfeng) [reply scripts/e2e/run-e2e.sh]: updated

File: scripts/e2e/run-e2e.sh (hunk 5 above)
Comment (3quanfeng) [reply scripts/e2e/run-e2e.sh]: removed

**Pull request merged**